tests/e2e/util.go

```diff
@@ -1264,6 +1275,54 @@ func waitVCenterServiceToBeInState(serviceName string, host string, state string
 	return waitErr
 }
 
+// checkVcenterServicesStatus checks and polls for vCenter essential services status
+// to be in running state
+func checkVcenterServicesRunning(host string, timeout ...time.Duration) error {
+	var pollTime time.Duration
+	if len(timeout) == 0 {
+		pollTime = pollTimeout * 2
+	} else {
+		pollTime = timeout[0]
+	}
+	waitErr := wait.PollImmediate(poll, pollTime, func() (bool, error) {
+		var runningServices []string
+		var statusMap = make(map[string]bool)
+		sshCmd := fmt.Sprintf("service-control --%s", statusOperation)
+		framework.Logf("Invoking command %v on vCenter host %v", sshCmd, host)
+		result, err := fssh.SSH(sshCmd, host, framework.TestContext.Provider)
+		if err != nil || result.Code != 0 {
+			fssh.LogResult(result)
+			return false, fmt.Errorf("couldn't execute command: %s on vCenter host: %v", sshCmd, err)
+		}
+		framework.Logf("Command %v output is %v", sshCmd, result.Stdout)
+
+		services := strings.SplitAfter(result.Stdout, svcRunningMessage+":")
+		stoppedService := strings.Split(services[1], svcStoppedMessage+":")
+		if strings.Contains(stoppedService[0], "StartPending:") {
+			runningServices = strings.Split(stoppedService[0], "StartPending:")
+		} else {
+			runningServices = append(runningServices, stoppedService[0])
+		}
+		essentialServices := []string{spsServiceName, vsanhealthServiceName, vpxdServiceName, wcpServiceName}
```

Comment:
Have an option where we can send additional/list of services to be checked.
We don't need wcp svc in vanilla setup for ex.